File: pizzaQueueJava/PizzaQueue.java

public class PizzaQueue
{
    Pizza head, tail;

    public PizzaQueue(Pizza head, Pizza tail)
    {
        this.head = head;
        this.tail = tail;
    }

    public void enqueue(Pizza thispizza)
    {
        if (head == null) 
        {head = thispizza;}
        else 
        {tail.next = thispizza;}
        tail = thispizza;
        return;
    }
    
    public Pizza dequeue()
    {
        Pizza pizzatodeliver = null;
        if (head != null)
        {
        	pizzatodeliver = head;
        	head = head.next;
        }
        if (head == null) 
        	tail = null;
        return pizzatodeliver;
    }
    
    public void deliver()
    {
        Pizza thispizza = this.dequeue();
        if (thispizza == null)
        {
        	System.out.println("No deliveries pending"); 
        	return;
        }
        System.out.println("Deliver a pizza with " + thispizza.toppings + " to " + thispizza.address);
    }
    
    static void onestore() {
    	Pizza first = null, last = null;
    	
        PizzaQueue myQueue = new PizzaQueue(first, last);
        
        myQueue.enqueue(new Pizza("pepperoni", "1234 Bobcat Trail"));
        myQueue.enqueue(new Pizza("sausage", "2345 University Drive"));
        myQueue.deliver();
        myQueue.enqueue(new Pizza("extra cheese", "3456 Rickster Road"));
        myQueue.enqueue(new Pizza("everything", "4567 King Court"));
        myQueue.enqueue(new Pizza("coffee beans", "5678 Java Circle"));
        myQueue.deliver();
        myQueue.deliver();
        myQueue.deliver();
        myQueue.deliver();
        myQueue.deliver();
    }
    
    static void twostores() {
    	Pizza s1f = null, s1l = null;
    	Pizza s2f = null, s2l = null;
    	
        PizzaQueue firstQueue = new PizzaQueue(s1f, s1l);
        PizzaQueue secondQueue = new PizzaQueue(s2f, s2l);
        
        firstQueue.enqueue(new Pizza("111111", "111111111111111"));
        secondQueue.enqueue(new Pizza("AAAAAA", "AAAAAAAAAAAAAAA"));
        firstQueue.enqueue(new Pizza("222222", "222222222222222"));
        secondQueue.enqueue(new Pizza("BBBBBB", "BBBBBBBBBBBBBBB"));
        firstQueue.enqueue(new Pizza("333333", "333333333333333"));
        secondQueue.enqueue(new Pizza("CCCCCC", "CCCCCCCCCCCCCCC"));
        firstQueue.enqueue(new Pizza("444444", "444444444444444"));
        secondQueue.enqueue(new Pizza("DDDDDD", "DDDDDDDDDDDDDDD"));
        firstQueue.enqueue(new Pizza("555555", "555555555555555"));
        secondQueue.enqueue(new Pizza("EEEEEE", "EEEEEEEEEEEEEEE"));
        firstQueue.enqueue(new Pizza("666666", "666666666666666"));
        secondQueue.enqueue(new Pizza("FFFFFF", "FFFFFFFFFFFFFFF"));
        	
        firstQueue.deliver();
        firstQueue.deliver();
        firstQueue.deliver();
        firstQueue.deliver();
        firstQueue.deliver();
        firstQueue.deliver();
        firstQueue.deliver();

        secondQueue.deliver();
        secondQueue.deliver();
        secondQueue.deliver();
        secondQueue.deliver();
        secondQueue.deliver();
        secondQueue.deliver();
        secondQueue.deliver();	
    }
    
    public static void main(String args[])
    {
    	onestore();
    	//twostores();
    }
}

File: pizzaQueue.cpp
#include <iostream>
#include <string>
using namespace std;

class pizza
{
public:
	string toppings, address;
	pizza* next;
	pizza(string toppings, string address)
	{
		this->address = address;
		this->toppings = toppings;
		next = NULL;
	}
};

void enqueue(pizza** head, pizza** tail, pizza* thispizza)
{
	if (*head == NULL) *head = thispizza;
	else (*tail)->next = thispizza;
	*tail = thispizza;
	return;
}

pizza* dequeue(pizza** head, pizza** tail)
{
	pizza* pizzatodeliver = NULL;
	if (*head != NULL)
	{
		pizzatodeliver = *head;
		*head = (*head)->next;
	}
	if (*head == NULL) *tail = NULL;
	return pizzatodeliver;
}

void deliver(pizza** head, pizza** tail)
{
	pizza* thispizza = dequeue(head, tail);
	if (thispizza == NULL)
	{
		cout << "No deliveries pending" << endl; return;
	}
	cout << "Deliver a pizza with " << thispizza->toppings
		 << " to " << thispizza->address << endl;
}

void onestore()
{
	pizza *first=NULL, *last=NULL;

	enqueue(&first, &last, new pizza("pepperoni", "1234 Bobcat Trail"));
	enqueue(&first, &last, new pizza("sausage", "2345 University Drive"));
	deliver(&first, &last);
	enqueue(&first, &last, new pizza("extra cheese", "3456 Rickster Road"));
	enqueue(&first, &last, new pizza("everything", "4567 King Court"));
	enqueue(&first, &last, new pizza("coffee beans", "5678 Java Circle"));
	deliver(&first, &last);
	deliver(&first, &last);
	deliver(&first, &last);
	deliver(&first, &last);
	deliver(&first, &last);
}

void twostores()
{
	pizza* s1f = NULL, * s1l = NULL;  //first and last ptrs for 1st store queue
	pizza* s2f = NULL, * s2l = NULL;  //first and last ptrs for 2nd store queue

	enqueue(&s1f, &s1l, new pizza("111111", "111111111111111"));
	enqueue(&s2f, &s2l, new pizza("AAAAAA", "AAAAAAAAAAAAAAA"));
	enqueue(&s1f, &s1l, new pizza("222222", "222222222222222"));
	enqueue(&s2f, &s2l, new pizza("BBBBBB", "BBBBBBBBBBBBBBB"));
	enqueue(&s1f, &s1l, new pizza("333333", "333333333333333"));
	enqueue(&s2f, &s2l, new pizza("CCCCCC", "CCCCCCCCCCCCCCC"));
	enqueue(&s1f, &s1l, new pizza("444444", "444444444444444"));
	enqueue(&s2f, &s2l, new pizza("DDDDDD", "DDDDDDDDDDDDDDD"));
	enqueue(&s1f, &s1l, new pizza("555555", "555555555555555"));
	enqueue(&s2f, &s2l, new pizza("EEEEEE", "EEEEEEEEEEEEEEE"));
	enqueue(&s1f, &s1l, new pizza("666666", "666666666666666"));
	enqueue(&s2f, &s2l, new pizza("FFFFFF", "FFFFFFFFFFFFFFF"));

	deliver(&s1f, &s1l);
	deliver(&s1f, &s1l);
	deliver(&s1f, &s1l);
	deliver(&s1f, &s1l);
	deliver(&s1f, &s1l);
	deliver(&s1f, &s1l);
	deliver(&s1f, &s1l);

	deliver(&s2f, &s2l);
	deliver(&s2f, &s2l);
	deliver(&s2f, &s2l);
	deliver(&s2f, &s2l);
	deliver(&s2f, &s2l);
	deliver(&s2f, &s2l);
	deliver(&s2f, &s2l);
}

int main()
{
	// onestore();
	twostores();
	return 0;
}